### PR TITLE
Droth 1696 hazmat prohibitions traffic signs update v3

### PR DESCRIPTION
**Pull request by oskarmalmstrom**

### CODE REVIEW

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/middleware/TrafficSignManager.scala [hunk 1]
```diff
@@ -32,23 +39,51 @@ object TrafficSignManager {
 case class TrafficSignManager(manoeuvreService: ManoeuvreService, roadLinkService: RoadLinkService) {
   def withDynTransaction[T](f: => T): T = OracleDatabase.withDynTransaction(f)
 
+  case object TrafficSignSerializer extends CustomSerializer[TrafficSignProperty](format =>
```
Comment (oskarmalmstrom) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/middleware/TrafficSignManager.scala]: can these serializers be put on the JsonSerializer.scala and then imported? this one seems to be identical with the other one

File: digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/asset/asset.scala [hunk 2]
```diff
@@ -331,20 +362,25 @@ object ProhibitionClass {
...
+//      } else Seq()
+//    )++
+//      (if (prohibitionValue.intersect(Seq(Pedestrian, Bicycle)).size == 2) {
+//        prohibitionValue --= NoCyclesOrMopeds.values
+//        Seq(NoCyclesOrMopeds)
+//      } else Seq()
+//        ) ++ prohibitionValue.flatMap { value =>
+//      ProhibitionClass.apply(value).trafficSign
+//    }
+//  }
```
Comment (MAC-CGI) [inline digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/asset/asset.scala]: should be deleted

File: digiroad2-oracle/src/main/resources/db/migration/V199__add_new_table_for_traffic_manager_process.sql [hunk 3]
```diff
@@ -2,6 +2,7 @@ CREATE TABLE	TRAFFIC_SIGN_MANAGER
 (
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/resources/db/migration/V199__add_new_table_for_traffic_manager_process.sql]: This file needs to be deleted, and this code should be moved to DROTH-1814_vehicle_restrictions_and_traffic_signs_V2

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/pointasset/OracleTrafficSignDao.scala [hunk 4]
```diff
@@ -414,16 +414,16 @@ object OracleTrafficSignDao {
...
+//
+//  def getTrafficSignType(id: Long): Option[Int]= {
+//    sql""" select ev.value
+//         from asset a
+//         join property p on a.asset_type_id = p.asset_type_id and p.public_id = 'trafficSigns_type'
+//         left join single_choice_value scv on scv.asset_id = a.id and scv.property_id = p.id and p.property_type = 'single_choice'
+//         left join enumerated_value ev on scv.enumerated_value_id = ev.id
+//         where a.asset_type_id = ${TrafficSigns.typeId} and a.id = $id
+//    """.as[Int].firstOption
+//  }
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/pointasset/OracleTrafficSignDao.scala]: could be deleted

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/HazmatTransportProhibitionService.scala [hunk 5]
```diff
@@ -90,4 +89,57 @@ class HazmatTransportProhibitionService(roadLinkServiceImpl: RoadLinkService, ev
...
+//      withDynTransaction {
+//        val assetIds = dao.getConnectedAssetFromTrafficSign(trafficSignId)
+//        dao.fetchProhibitionsByIds(HazmatTransportProhibition.typeId, assetIds.toSet)
+//      }
+//    } else {
+//      val assetIds = dao.getConnectedAssetFromTrafficSign(trafficSignId)
+//      dao.fetchProhibitionsByIds(HazmatTransportProhibition.typeId, assetIds.toSet)
+//    }
+//    assets
+//  }
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/HazmatTransportProhibitionService.scala]: should be deleted

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/LinearAssetService.scala [hunk 6]
```diff
@@ -829,27 +829,25 @@ trait LinearAssetOperations {
...
-  }
+//  def createBasedOnTrafficSign(trafficSignInfo: TrafficSignInfo, newTransaction: Boolean = true): Seq[Long] = {
+//    if(newTransaction) {
+//      withDynTransaction {
+//        createLinearAssetFromTrafficSign(trafficSignInfo)
+//      }
+//    }
+//    else
+//      createLinearAssetFromTrafficSign(trafficSignInfo)
+//  }
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/LinearAssetService.scala]: should be deleted

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TrafficSignLinearGenerator.scala [hunk 7]
```diff
@@ -30,11 +34,36 @@ trait TrafficSignLinearGenerator {
 
   def withDynSession[T](f: => T): T = OracleDatabase.withDynSession(f)
 
+  //  type AssetValue <: Value
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TrafficSignLinearGenerator.scala]: delete

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TrafficSignLinearGenerator.scala [hunk 8]
```diff
@@ -365,7 +401,7 @@ trait TrafficSignLinearGenerator {
     newSeg.signId.diff(oldSeg.signId).foreach(sign => createAssetRelation(oldSeg.oldAssetId.get, sign))
   }
 
-  def combine(segments: Seq[TrafficSignToLinear]): Seq[TrafficSignToLinear] = {
+  def combine(segments: Seq[TrafficSignToLinear] /*, endRoadLinksInfo: Seq[(RoadLink, Option[Point], Option[Point])]*/): Seq[TrafficSignToLinear] = {
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TrafficSignLinearGenerator.scala]: delete

File: digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TrafficSignLinearGenerator.scala [hunk 9]
```diff
@@ -379,7 +415,7 @@ trait TrafficSignLinearGenerator {
       }
     }
 
-    def combineEqualValues(segmentPieces: Seq[TrafficSignToLinear]): Seq[TrafficSignToLinear] = {
+    def combineEqualValues(segmentPieces: Seq[TrafficSignToLinear] /*, segments : Seq[TrafficSignToLinear]*/): Seq[TrafficSignToLinear] = {
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/TrafficSignLinearGenerator.scala]: delete

File: digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala [hunk 10]
```diff
@@ -1730,100 +1727,100 @@ object DataFixture {
...
-        trafficSigns.map {trafficSign =>
-          println("traffic ->  type: " + trafficSignService.getProperty(trafficSign, trafficSignService.typePublicId) + " linkId: " + trafficSign.linkId + " position: " + trafficSign.lon + ", " + trafficSign.lat)
-          trafficSignService.create(trafficSign, username, roadLink)
-        }
-      }
-      println("")
-      println("Complete at time: " + DateTime.now())
-    }
-  }
+//  def createTrafficSignsUsingLinearAssets(): Unit = {
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala]: should be deleted

File: digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala [hunk 11]
```diff
@@ -1929,8 +1926,8 @@ object DataFixture {
             mergeAdditionalPanelsToTrafficSigns(trafficSignGroup(group))
           case _ => println("Please provide a traffic sign group")
         }
-      case Some("create_traffic_signs_using_linear_assets") =>
-        createTrafficSignsUsingLinearAssets()
+//      case Some("create_traffic_signs_using_linear_assets") =>
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala]: should be deleted

File: digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala [hunk 12]
```diff
@@ -1956,8 +1955,8 @@ object DataFixture {
         " verify_inaccurate_speed_limit_assets | update_information_source_on_existing_assets  | update_traffic_direction_on_roundabouts |" +
         " update_information_source_on_paved_road_assets | import_municipality_codes | update_municipalities | remove_existing_trafficSigns_duplicates |" +
         " create_manoeuvres_using_traffic_signs | merge_additional_panels_to_trafficSigns | create_traffic_signs_using_linear_assets | " +
-        " update_floating_stops_on_terminated_roads | update_private_roads | add_geometry_to_linear_assets | create_prohibition_using_traffic_signs |" +
-        " merge_additional_panels_to_trafficSigns | create_traffic_signs_using_linear_assets | create_prohibitions_using_traffic_signs")
+        " update_floating_stops_on_terminated_roads | update_private_roads | add_geometry_to_linear_assets | create_prohibition_using_traffic_signs | " +
```
Comment (MAC-CGI) [inline digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/DataFixture.scala]: create_traffic_signs_using_linear_assets  should be delete as suggestion

File: digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/TrafficSignLinearGeneratorSpec.scala [hunk 13]
```diff
@@ -382,5 +383,38 @@ class TrafficSignLinearGeneratorSpec extends FunSuite with Matchers {
...
+    val hazmatResultB = hazmatTransportProhibitionGenerator.createValue(Seq(trafficSignB))
+    hazmatResultB should be (Some(hazmatValueB))
+
+    val signProp = Seq(TrafficSignProperty(0, "trafficSigns_type", "", false, Seq(TextPropertyValue(NoVehiclesWithDangerGoods.OTHvalue.toString))))
+    val trafficSign = PersistedTrafficSign(1, 1000, 100, 0, 50, false, 0, 235, signProp, None, None, None, None, SideCode.AgainstDigitizing.value, None, NormalLinkInterface)
+    hazmatTransportProhibitionGenerator.createValue(Seq(trafficSign)).isEmpty should be (true)
   }
 
+
+  test("test deserialized"){
```
Comment (marquesfm) [inline digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/util/TrafficSignLinearGeneratorSpec.scala]: Empty test, to be removed

**Pull request merged**